libs/wingsdk/src/fs/fs.ts

```diff
@@ -0,0 +1,270 @@
+ * and will be normalized to the target platform if running on Windows.
+ */
+export class Util {
+  /**
+   * Join all arguments together and normalize the resulting path.
+   * @param paths The array of path need to join.
+   * @returns The resulting path after joining all the paths.
+   */
+  public static join(...paths: string[]): string {
+    return normalPath(path.join(...paths));
+  }
+  /**
+   * Retrieve the name of the directory from a given file path.
+   * @param p The path to evaluate.
+   * @returns The directory name of the path.
+   */
+  public static dirname(p: string): string {
+    return normalPath(path.dirname(p));
+  }
+  /**
+   * Retrieve the final segment of a given file path.
+   * @param p The path to evaluate.
+   * @returns The last portion of a path.
+   */
+  public static basename(p: string): string {
+    return path.basename(p);
+  }
+  /**
+   * Solve the relative path from {from} to {to} based on the current working directory.
+   * At times we have two absolute paths, and we need to derive the relative path from one to the other.
+   * @returns The relative path from {from} to {to}.
+   */
+  public static relative(from: string, to: string): string {
+    return normalPath(path.relative(from, to));
+  }
+  /**
+   * The right-most parameter is considered {to}. Other parameters are considered an array of {from}.
+   *
+   * Starting from leftmost {from} parameter, resolves {to} to an absolute path.
+   *
+   * If {to} isn't already absolute, {from} arguments are prepended in right to left order,
+   * until an absolute path is found. If after using all {from} paths still no absolute path is found,
+   * the current working directory is used as well. The resulting path is normalized,
+   * and trailing slashes are removed unless the path gets resolved to the root directory.
+   *
+   * @param paths A sequence of paths or path segments.
+   * @returns The resulting path after performing the resolve operation.
+   */
+  public static resolve(...paths: string[]): string {
+    return normalPath(path.resolve(...paths));
+  }
+  /**
+   * Check if the path exists.
+   * @param p The path to evaluate.
+   * @returns `true` if the path exists, `false` otherwise.
+   */
+  public static exists(p: string): boolean {
+    return fs.existsSync(p);
+  }
+  /**
+   * Read the contents of the directory.
+   * @param dirpath The path to evaluate.
+   * @returns The contents of the directory.
+   */
+  public static readdir(dirpath: string): Array<string> {
+    return fs.readdirSync(dirpath);
+  }
+  /**
+   * If the path exists, read the contents of the directory; otherwise, return `undefined`.
+   * @param dirpath The path to evaluate.
+   * @returns The contents of the directory if the path exists, `undefined` otherwise.
+   */
+  public static tryReaddir(dirpath: string): Array<string> | undefined {
+    if (!Util.exists(dirpath)) {
+      return undefined;
+    }
+    return Util.readdir(dirpath);
```

Comment:
No need to call exists first. readdir can fail for other reasons so might as well make this more bulletproof.

```suggestion
    try {
      return Util.readdir(dirpath);
    } catch {
      return undefined;
    }
```